dspace-uclouvain/src/main/java/org/dspace/uclouvain/pdfAttestationGenerator/rest/PDFAttestationGeneratorRestController.java

```diff
@@ -0,0 +1,87 @@
+package org.dspace.uclouvain.pdfAttestationGenerator.rest;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.UUID;
+
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.fop.apps.FOPException;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.Item;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.uclouvain.pdfAttestationGenerator.exceptions.PDFGenerationException;
+import org.dspace.uclouvain.pdfAttestationGenerator.factory.PDFAttestationGeneratorFactory;
+import org.dspace.uclouvain.pdfAttestationGenerator.handlers.PDFAttestationGeneratorHandler;
+import org.dspace.web.ContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.xml.sax.SAXException;
+
+@RestController
+@RequestMapping("/api/uclouvain/item/{uuid}/attestation")
+public class PDFAttestationGeneratorRestController {
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    /** 
+     * Generates and returns a PDF attestation with a template depending on the targeted DSpace object type
+     * 
+     * @param uuid: The uuid of the target DSpace Item from the request
+     */
+    @RequestMapping(method = RequestMethod.GET)
+    public void attestation(
+        HttpServletResponse response,
+        HttpServletRequest request,
+        @PathVariable UUID uuid
+    ) throws FileNotFoundException, IOException, FOPException, TransformerConfigurationException, TransformerException, SAXException, SQLException {
+        try {
+            PDFAttestationGeneratorHandler handler = PDFAttestationGeneratorFactory.getInstance().getHandlerInstance(uuid);
+            if (handler == null) {
+                response.sendError(400, "Object not found or not handled");
+            }
```

Comment:
This is exactly what I explain : You received `null` instead of a specific exception and you cannot know the initial problem. Dealing with exception, you could be more specific here
```java
try {
    PDFAttestationGeneratorHandler handler = PDFAttestationGeneratorFactory.getInstance().getHandlerInstance(uuid);
} catch (HandleNotFoundException hnfe) {
    response.sendError(400, "Handle not found ....");
    return;
} catch (SQLException sqle) {
    response.sendError(404, "Object not found");
    return;
}
```